### ISSUE TITLE
fixed_regex_linter incorrectly suggest that my regex with "\>" is static

### ISSUE THREAD

**Issue by dschlaep** [r-lib/lintr#1478]
My code removes some text from the end of some other text using `sub()`. 

`fixed_regex_linter()` (incorrectly) suggests that my regular expression is static (example 1 below) if I use the symbol "\\>" to match the empty string at end of a word. The suggestion with `fixed = TRUE` does not reproduce my desired outcome. However, `fixed_regex_linter()` correctly recognizes that regular expressions with "$" and "\b" are not static (examples 2 and 3) - obviously, I could use one of those instead.

I am not sure whether this has anything to do with the fact that regular expressions with "$" and "\b" work for both `perl = FALSE` and `perl = TRUE` while those with "\\>" work only for `perl = FALSE` (?regex).

#### Example: trim "_stuff" only from end of "name_stuff_more_stuff" with desired result "name_stuff_more"

``` r

# 1) Approach "\>"
print(sub("_stuff\\>", "", "name_stuff_more_stuff"))
#> [1] "name_stuff_more"

# Lint
lintr::lint(
  text = 'sub("_stuff\\\\>", "", "name_stuff_more_stuff")',
  linters = lintr::fixed_regex_linter()
)
#> <text>:1:5: warning: [fixed_regex_linter] This regular expression is static, i.e., its matches can be expressed as a fixed substring expression, which is faster to compute. Here, you can use "_stuff>" with fixed = TRUE.
#> sub("_stuff\\>", "", "name_stuff_more_stuff")
#>     ^~~~~~~~~~~

# Suggestion by `fixed_regex_linter()` is incorrect
print(sub("_stuff>", "", "name_stuff_more_stuff", fixed = TRUE))
#> [1] "name_stuff_more_stuff"
```

<sup>Created on 2022-07-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

``` r
# 2) Approach "$"
print(sub("_stuff$", "", "name_stuff_more_stuff"))
#> [1] "name_stuff_more"

# No lint
lintr::lint(
  text = 'sub("_stuff$", "", "name_stuff_more_stuff")',
  linters = lintr::fixed_regex_linter()
)
```

<sup>Created on 2022-07-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

``` r
# 3) Approach "\b"
print(sub("_stuff\\b", "", "name_stuff_more_stuff"))
#> [1] "name_stuff_more"

# No lint
lintr::lint(
  text = 'sub("_stuff\\\\b", "", "name_stuff_more_stuff")',
  linters = lintr::fixed_regex_linter()
)
```

<sup>Created on 2022-07-27 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

``` r
print(packageVersion("lintr"))
#> [1] '3.0.0.9000'
```

**Comment by MichaelChirico** [comment]
Note some discussion here BTW:

https://github.com/r-lib/lintr/pull/1032#issuecomment-1129325440

I think at root we want to detect the value of `perl=` and tweak the logic for what's a fixed regex or not accordingly. There are very subtle (and rare! `\<` is almost unheard of 😸) differences.

**Comment by janlimbeck** [comment]
Any updates on when this issue might be resolved?

**Comment by MichaelChirico** [comment]
Hi @janlimbeck, we're all volunteers to if this affects you a lot, feel free to take it up & go for a fix! PRs welcome; happy to help you along if you're stuck.
